packages/oracle-node/src/fetchers/api-fetcher/ApiFetcher.ts

```diff
@@ -0,0 +1,76 @@
+import axios from "axios";
+import jp from "jsonpath";
+import Decimal from "decimal.js";
+import {
+  MultiRequestFetcher,
+  RequestIdToResponse,
+} from "../MultiRequestFetcher";
+import { getLastPriceOrFail } from "../../db/local-db";
+
+interface ApiFetcherConfig {
+  url: string;
+  jsonpath: string;
+  pairedToken?: string;
+  isReverted?: boolean;
+}
+
+export class ApiFetcher extends MultiRequestFetcher {
+  protected override retryForInvalidResponse: boolean = true;
+
+  constructor(
+    apiName: string,
+    private apiFetcherConfig: Record<string, ApiFetcherConfig>
+  ) {
+    super(`${apiName}-api`);
+  }
+
+  override async makeRequest(dataFeedId: string): Promise<unknown> {
+    const { url } = this.apiFetcherConfig[dataFeedId];
+    return (await axios.get<unknown>(url)).data;
+  }
+
+  override extractPrice(
+    dataFeedId: string,
+    responses: RequestIdToResponse<unknown>
+  ): number | undefined {
+    const response = responses[dataFeedId];
+    const rawPrice = this.extractPriceFromResponse(dataFeedId, response);
+    return this.calculatePriceBasedOnConfig(dataFeedId, rawPrice).toNumber();
+  }
+
+  private extractPriceFromResponse(
+    dataFeedId: string,
+    response: unknown
+  ): Decimal {
+    const { url, jsonpath } = this.apiFetcherConfig[dataFeedId];
+    const [extractedValue] = jp.query(response, jsonpath) as Decimal.Value[];
+
+    if (typeof extractedValue === "undefined") {
```

Comment:
```suggestion
    if (xtractedValue === undefined) {
```
Is simpler, but does the same, right?

Reply:
![image](https://github.com/redstone-finance/redstone-oracles-monorepo/assets/35804520/98f0d412-9b0f-426b-8a95-169bf2541595)